vertx-db2-client/src/test/java/io/vertx/db2client/tck/DB2BinaryDataTypeDecodeTest.java

```diff
@@ -61,7 +65,31 @@ public void testDouble(TestContext ctx) {
     }
 
     // For DB2/z the largest value that can be stored in a DOUBLE column is 7.2E75
-    testDecodeGeneric(ctx, "test_float_8", Double.class, (double) 7.2E75);
+    testDecodeGeneric(ctx, "test_float_8", Double.class, JDBCType.DOUBLE, (double) 7.2E75);
+  }
+  
+  @Override
+  public void testChar(TestContext ctx) {
+    // Override to expecting JDBCType.CHAR instead of VARCHAR
+    testDecodeGeneric(ctx, "test_char", String.class, JDBCType.CHAR, "testchar");
+  }
+  
+  @Override
+  public void testNumeric(TestContext ctx) {
+    // Override to expecting JDBCType.DECIMAL instead of NUMERIC
+    testDecodeGeneric(ctx, "test_numeric", Numeric.class, JDBCType.DECIMAL, Numeric.parse("999.99"));
+  }
+  
+  @Override
+  public void testDecimal(TestContext ctx) {
+    // Override to expecting JDBCType.DECIMAL instead of NUMERIC
+    testDecodeGeneric(ctx, "test_decimal", Numeric.class, JDBCType.DECIMAL, Numeric.parse("12345"));
+  }
+  
+  @Override
+  public void testTime(TestContext ctx) {
+    // Override to expecting JDBCType.TIME instead of DATE
+    testDecodeGeneric(ctx, "test_time", LocalTime.class, JDBCType.TIME, LocalTime.of(18, 45, 2));
```

Comment:
I'm still a bit unclear on how the `JDBCType` will be used. Is it OK that I'm overriding some of the expected JDBCTypes for DB2 here? Or is it important that all DBs are consistent with one another for this use case? CC @gavinking

Reply:
Well we’re going to use it to decide what kind of Java object to read from that column of the result set. For example, if you return `Types.VARCHAR`, we will read strings.

Reply:
So for `testChar`, in DB2 the column is defined as `CHAR(6)`. To me it seemed correct that we classify that as `JDBCType.CHAR`, but since a char[6] is basically the same thing as `VARCHAR/String`, should I have it return `JDBCType.VARCHAR` for a `CHAR(6)` column?

Reply:
I would just try to copy what the JDBC driver does.

Reply:
@gavinking what can we assume among these ?

- same java type => the same SQL type?
- same java type <= the same SQL type?
- same java type <=> the same SQL type?
- no relationship ?

Reply:
No it's nonsimple. If you look at the appendices of the JDBC spec you'll see that it's a many-to-many mapping between Java types and SQL spec types. And then the Hibernate built-in types only loosely follow that mapping.

But that's not what we're trying to capture here. What we're going for here is a mapping from database types to equivalent ANSI SQL types. Now that is also nonsimple, but it's *simpler*, and we can look at what mapping the JDBC drivers use for inspiration.

Reply:
(But of course Hibernate does expect some consistency: if you return `TIMESTAMP`, it's going to expect to get a `LocalDateTime` when it reads the result set, and if you return `TIMESTAMP_WITH_TIMEZONE`, it's going to expect to get an `OffsetDateTime`.)

Reply:
ok